cpp/ql/src/semmle/code/cpp/ir/implementation/MemoryAccessKind.qll

```diff
@@ -1,10 +1,15 @@
 import cpp
 
-newtype TMemoryAccessKind =
+private newtype TMemoryAccessKind =
   TIndirectMemoryAccess() or
+  TIndirectMayMemoryAccess() or
+  TBufferMemoryAccess() or
+  TBufferMayMemoryAccess() or
   TEscapedMemoryAccess() or
   TPhiMemoryAccess() or
-  TUnmodeledMemoryAccess()
+  TUnmodeledMemoryAccess() or
+  TChiOldMemoryAccess() or
+  TChiUpdateMemoryAccess()
```

Comment:
From my corresponding comment on #172:

> These new memory access kinds seem to closely mirror the new opcodes, which again are mirrored by the new instruction classes. Is all this redundancy really needed? I'm probably asking this because I still don't feel I understand memory access kinds well, especially result memory access kinds.

Reply:
I've added a big QLDoc comment to `ChiInstruction`.